dataclients/kubernetes/clusterclient.go

```diff
@@ -430,9 +444,22 @@ func (c *clusterClient) fetchClusterState() (*clusterState, error) {
 		err         error
 		ingressesV1 []*definitions.IngressV1Item
 		ingresses   []*definitions.IngressItem
+		secrets     []*secret
 	)
 	if c.ingressV1 {
 		ingressesV1, err = c.loadIngressesV1()
+		for _, ing := range ingressesV1 {
+			if ing.Spec.IngressTLS != nil {
+				for _, sec := range ing.Spec.IngressTLS {
+					secret, err := c.loadSecret(sec.SecretName, ing.Metadata.Namespace)
+					if err != nil {
+						log.Println(err)
+						return nil, err
+					}
+					secrets = append(secrets, secret)
+				}
+			}
+		}
```

Comment:
* this should better reside in the separate method e.g. `c.loadTLSSecrets(ingressesV1)`
* several ingresses may reuse the same secret so the code should load each `namespace+secret name` only once
* errors should be handled gracefully instead of early return, otherwise single malformed ingress renders whole cluster non-functioning
* specific `log.Error` or `log.Warn` should be used instead of `log.Println` like elsewhere